http-api/src/error.rs

```diff
@@ -28,29 +28,45 @@ pub enum Error {
     #[error("could not resolve head: {0}")]
     NoHead(&'static str),
 
-    /// An error occured with radicle identities.
+    /// An error occurred during an authentication process.
+    #[error("could not authenticate: {0}")]
+    Auth(&'static str),
+
+    /// An error occurred while verifying the siwe message.
+    #[error(transparent)]
+    SiweVerificationError(#[from] siwe::VerificationError),
+
+    /// An error occurred while parsing the siwe message.
+    #[error(transparent)]
+    SiweParseError(#[from] siwe::ParseError),
```

Comment:
This seems maybe overkill but it's the only way I found to match against the `siwe` crate errors